fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -0,0 +1,192 @@
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @throws IllegalArgumentException thrown if JSON object `fields` is not present
+     * @throws IOException thrown if there is error parsing the avro files
+     */
+    public void parseAvroFileDimensionRows(Dimension dimension, String avroFilePath, String avroSchemaPath)
+        throws IllegalArgumentException, IOException {
+
+        if (isSchemaValid(dimension, avroSchemaPath)) {
+
+            // Creates an AVRO schema object based on the parsed AVRO schema file (.avsc)
+            Schema schema = new Schema.Parser().parse(new File(avroSchemaPath));
+
+            // Creates an AVRO DatumReader object based on the AVRO schema object
+            DatumReader<GenericRecord> datumReader = new GenericDatumReader<>(schema);
+
+            // Creates an AVRO DataFileReader object that reads the AVRO data file one record at a time
+            DataFileReader<GenericRecord> dataFileReader = new DataFileReader<>(new File(avroFilePath), datumReader);
+
+            // Generates a set of dimension Rows after retrieving the appropriate fields
+            Set<DimensionRow> dimensionRows = StreamSupport.stream(dataFileReader.spliterator(), false)
+                    .map(genericRecord -> dimension.getDimensionFields().stream().collect(
+                             Collectors.toMap(
+                                 DimensionField::getName,
+                                 dimensionField -> genericRecord.get(
+                                     dimensionFieldMapper.convert(dimension, dimensionField)
+                                 ).toString()
+                             )
+                         )
+                    )
+                    .map(dimension::parseDimensionRow)
+                    .collect(Collectors.toSet());
+
+                // Adds all dimension rows to the dimension cache
+                dimension.addAllDimensionRows(dimensionRows);
+        } else {
+            String msg = "The AVRO schema file is corrupted / empty {}";
+            LOG.error(msg);
+            throw new IllegalArgumentException(msg);
+        }
+    }
+
+    /**
+     * Constructs a JSON Node object from the avro schema file.
+     *
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @return JsonNode object
+     *
+     * @throws IOException thrown if there is error parsing the avro files
+     */
+    private JsonNode convertFileToJsonNode(String avroSchemaPath) throws IOException {
+
+        try {
+            File file = new File(avroSchemaPath);
+            JsonNode jsonNode = objectMapper.readTree(file);
+            return jsonNode;
+        } catch (JsonProcessingException e) {
+            LOG.error("Unable to process the Json {}", e);
+            throw e;
+        } catch (IOException e) {
+            LOG.error("Unable to process the file {}", e);
+            throw e;
```

Comment:
I segregated them because they are 2 different exceptions, and I thought they need to be logged differently. Let me, know if I can combine them into a single handler with a generic error message '''`Unable to process the file {}`'''